plugins/web/opentelemetry-plugin-react-load/src/BaseOpenTelemetryComponent.ts

```diff
@@ -0,0 +1,541 @@
+          }
+        );
+        // if shouldComponentUpdate returns false, the component does not get
+        // updated and no other lifecycle methods get called
+        if (!apply) {
+          plugin._endParentSpan(this);
+        }
+
+        return apply;
+      };
+    };
+  }
+
+  /**
+   * Patches the shouldComponentUpdate lifecycle method
+   */
+  private _patchGetSnapshotBeforeUpdate() {
+    return (
+      original: GetSnapshotBeforeUpdateFunction
+    ): GetSnapshotBeforeUpdateFunction => {
+      const plugin = this;
+
+      return function patchGetSnapshotBeforeUpdate(
+        this: React.Component,
+        ...args
+      ): any {
+        const parentData = plugin._getCreateParentSpan(this);
+        if (parentData) {
+          parentData.name = AttributeNames.UPDATING_SPAN;
+        }
+        return plugin._instrumentFunction(
+          this,
+          'getSnapshotBeforeUpdate',
+          () => {
+            return original!.apply(this, args);
+          }
+        );
+      };
+    };
+  }
+
+  /**
+   * Patches the componentDidUpdate lifecycle method
+   */
+  private _patchComponentDidUpdate() {
+    return (
+      original: ComponentDidUpdateFunction
+    ): ComponentDidUpdateFunction => {
+      const plugin = this;
+
+      return function patchComponentDidUpdate(
+        this: React.Component,
+        ...args
+      ): void {
+        const parentData = plugin._getCreateParentSpan(this);
+        if (parentData) {
+          parentData.name = AttributeNames.UPDATING_SPAN;
+        }
+        const apply = plugin._instrumentFunction(
+          this,
+          'componentDidUpdate',
+          () => {
+            return original!.apply(this, args);
+          }
+        );
+        plugin._endParentSpan(this);
+        return apply;
+      };
+    };
+  }
+
+  /**
+   * Patches the componentWillUnmount lifecycle method
+   */
+  private _patchComponentWillUnmount() {
+    return (
+      original: ComponentWillUnmountFunction
+    ): ComponentWillUnmountFunction => {
+      const plugin = this;
+
+      return function patchComponentWillUnmount(
+        this: React.Component,
+        ...args
+      ): void {
+        const parentData = plugin._getCreateParentSpan(this);
+        if (parentData) {
+          parentData.name = AttributeNames.UNMOUNTING_SPAN;
+        }
+        const apply = plugin._instrumentFunction(
+          this,
+          'componentWillUnmount',
+          () => {
+            return original!.apply(this, args);
+          }
+        );
+        plugin._endParentSpan(this);
+        return apply;
+      };
+    };
+  }
+
+  /**
+   * patch function which wraps all the lifecycle methods
+   */
+  public patch(): void {
+    BaseOpenTelemetryComponent._logger.debug(
+      'applying patch to',
+      this.moduleName,
+      this.version
+    );
+
+    if (isWrapped(this.render)) {
+      shimmer.unwrap(this, 'render');
+      BaseOpenTelemetryComponent._logger.debug(
```

Comment:
These seem like they should be warnings. If double patching is attempted, then something probably isn't right.